### ISSUE TITLE
When using voiceover to navigate the document page, the focus on the links is not logical.

### ISSUE THREAD

**Issue by msftedad** [gdh1995/pdf.js#22]
Attach (recommended) or Link to PDF file here:
https://mozilla.github.io/pdf.js/web/viewer.html.
[edad (2).pdf](https://github.com/gdh1995/pdf.js/files/14600836/edad.2.pdf)

Configuration:
- Operating system and its version: iPhone 13 Version-17.4
- PDF.js version: 1.4   

Pre-requisites:
Open Settings> Accessibility> Voiceover> Turn on Voiceover.

Steps to reproduce the problem:

1. Open this PDF viewer https://mozilla.github.io/pdf.js/web/viewer.html.
2. Navigate to 'Tools' option at header at header of pdf viewer.
3. Activate 'Open' control and upload any pdf (Having links in it) from your system.
4. Now you'll have the pdf file, in pdf viewer.
5. Navigate through the links and observe the issue.
6. 

What is the expected behavior? (add screenshot)
Voiceover focus should be logical and visible on the links when user navigates using swipe gestures and must announce the complete link information on the document page.

What went wrong? (add screenshot)
When using voiceover to navigate the document page, the focus on the links is not logical.

When using voice over to navigate the links, the focus order is not visible on the selected link. Instead, it moves to the beginning of the document page and announces it as a link.

https://github.com/gdh1995/pdf.js/assets/93735775/d31866ea-5b2e-4ed5-ab74-3faabc2ca34c

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension):

**Comment by msftedad** [comment]
Hi @gdh1995, Any update on this issue? Please provide the ETA to fix this issue.
